pkgs/applications/networking/avalanchego/default.nix

```
@@ -0,0 +1,52 @@
+{ buildGoModule
+, fetchFromGitHub
+, fetchpatch
+, lib
+, stdenv
+, IOKit
+}:
+
+buildGoModule rec {
+  pname = "avalanchego";
+  version = "1.9.5";
+
+  src = fetchFromGitHub {
+    owner = "ava-labs";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-Uim2Q3LvNEV0C5/gFdAiXKy2oKbhhc/ifRyz5BIh6zE=";
+  };
+buildInputs = lib.optionals stdenv.isDarwin [ IOKit ];
+
+  # patch zondax/hid dependency manually on darwin
+  patches = [(fetchpatch {
+    url = "https://github.com/ava-labs/avalanchego/pull/1194.patch";
+    name = "update-zondax-hid.patch";
+    hash = "";
+  })];
```

```suggestion
  patches = lib.optionals stdenv.isDarwin [
    (fetchpatch {
      url = "https://github.com/ava-labs/avalanchego/pull/1194.patch";
      name = "update-zondax-hid.patch";
      hash = "sha256-ylL++zrXG9flGzMvSC6CgaXO4+Lnwi7AU3Zt3QFfZEo=";
    })
  ];
```

Only apply patch on darwin.